p2p/src/channels/snark_effectful/p2p_channels_snark_effectful_actions.rs

```diff
@@ -0,0 +1,31 @@
+use crate::{channels::P2pChannelsEffectfulAction, P2pState, PeerId};
+use openmina_core::{snark::SnarkInfo, ActionEvent};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Debug, Clone, ActionEvent)]
+#[action_event(fields(display(peer_id)))]
+pub enum P2pChannelsSnarkEffectfulAction {
+    Init {
+        peer_id: PeerId,
+    },
+    RequestSend {
+        peer_id: PeerId,
+        limit: u8,
+    },
+    ResponseSend {
+        peer_id: PeerId,
+        snarks: Vec<SnarkInfo>,
+    },
+}
+
+impl redux::EnablingCondition<P2pState> for P2pChannelsSnarkEffectfulAction {
+    fn is_enabled(&self, _state: &P2pState, _time: redux::Timestamp) -> bool {
+        true
+    }
+}
+
+impl From<P2pChannelsSnarkEffectfulAction> for crate::P2pAction {
+    fn from(action: P2pChannelsSnarkEffectfulAction) -> Self {
+        Self::ChannelsEffectful(P2pChannelsEffectfulAction::Snark(action))
```

Comment:
Can we change `Self` here for the action type?

Reply:
@0xMimir one clarification, the problem is not the `Self` in the return, is the `Self` in the body. But I don't think it is an issue here, it is just in dispatchers and effect handlers (when `Self::` or aliases are used it makes it harder to grep for the action). I would revert to the original.

@dkuehr agree?

Reply:
Yes I meant in the return, but is not really an issue here. It just makes things easier to grep